DAQ_System/Source/main.cpp

```diff
@@ -55,41 +46,41 @@ int main()
     printf("Mounting the filesystem... ");
     fflush(stdout);
 
-    int err = fileSystem.mount(&blockDevice);
-    printf("%s\n", (err ? "Fail :(" : "OK"));
-    if (err) {
+    int status = file_system.mount(&block_device);
+    printf("%s\n", (status ? "Fail :(" : "OK"));
+    if (status) {
         // Reformat if we can't mount the filesystem
         // this should only happen on the first boot
         printf("No filesystem found, formatting... ");
         fflush(stdout);
-        err = fileSystem.reformat(&blockDevice);
-        printf("%s\n", (err ? "Fail :(" : "OK"));
-        if (err) {
-            error("error: %s (%d)\n", strerror(-err), err);
+        status = file_system.reformat(&block_device);
+        printf("%s\n", (status ? "Fail :(" : "OK"));
+        if (status) {
+            error("error: %s (%d)\n", strerror(-status), status);
         }
     }
```

Comment:
Modularize to a `mount()` function. The file system class already has its own `.mount()` method, but I would like to include error checking. Note that their `.reformat()` method uses the base class BlockDevice's `.init()` method, which explains why we do not have to call it ourselves when not using the file system class.